File: implementUnionFind.py
class UnionFind:
	def __init__(self, n, parent=None, size=None):
		self.parent=[i for i in range(n)]
		self.size=[1]*n

	def find(self, vertex):
		if self.parent[vertex]!=vertex:
			self.parent[vertex]=self.find(self.parent[vertex])

		return self.parent[vertex]
	
	def union(self, v1, v2):
		v1=self.find(v1)
		v2=self.find(v2)

		if v1!=v2:
			if self.size[v1]>self.size[v2]:
				self.parent[v2]=v1
				self.size[v1]+=self.size[v2]
			else:
				self.parent[v1]=v2
				self.size[v2]+=self.size[v1]
		return   

#un=UnionFind(2)

File: implementUnionFind.java
import java.util.*;
class UnionFind{
	int[] parent;
	int[] size;

	UnionFind(int n){
		this.parent=new int[n];
		this.size=new int[n];

		for(int i=0; i<n; i++){
			parent[i]=i;
			size[i]=1;
		}
	}

	public int find(int vertex){
		if(this.parent[vertex]!=vertex){
			this.parent[vertex]=find(this.parent[vertex]);
		}
		return this.parent[vertex];
	}

	public void union(int v1, int v2){
		v1=find(v1);
		v2=find(v2);

		if(v1!=v2){
			if(this.size[v1]>this.size[v1]){
				this.parent[v2]=v1;
				size[v1]+=size[v2];
			} else {
				parent[v1]=v2;
				size[v2]+=size[v1];
			}
		}
		return;
	}
}

class implementUnionFind{
	public static void main(String[] args) {
		UnionFind uf=new UnionFind(5);
	}
}